File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        res = 0
        l, r = 0, len(height) - 1

        while l < r:
            w = r - l
            h = min(height[l], height[r])
            res = max(res, w * h)

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return res
        
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0, w = 0, h = 0;
        int l = 0;
        int r = height.size() - 1;
        while (l < r) {
            w = r - l;
            h = min(height[l], height[r]);
            res = max(res, w * h);

            if (height[l] < height[r])
                l++;
            else
                r--;
        }
        return res;
    }
};